cmd/xl-v1-object.go

```diff
@@ -198,8 +198,14 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64, length i
 		return traceError(InvalidRange{startOffset, length, xlMeta.Stat.Size})
 	}
 
+	// Calculate endOffset according to length
```

Comment:
sorry, the loop is entered but we break because of `if length == totalBytesRead {` check.